virtualizarr/manifests/store.py

```diff
@@ -285,10 +312,44 @@ def list_prefix(self, prefix: str) -> AsyncGenerator[str, None]:
         raise NotImplementedError
 
     async def list_dir(self, prefix: str) -> AsyncGenerator[str, None]:
-        # docstring inherited
-        yield "zarr.json"
-        for k in self._group.arrays.keys():
-            yield k
+        # docstring
+        # Navigate to the target node
+        if not prefix:
+            node: ManifestArray | ManifestGroup = self._group
+            suffix = ""
+        else:
+            node, suffix = _get_deepest_group_or_array(self._group, prefix)
+        # Zarr-Python lists using a per-path basis, so we don't have anything to list
+        # as long as there is a suffix remaining and we require a '.' chunk separator in the ManifestArrays
+        if suffix:
+            return
+        # List contents based on node type
+        if isinstance(node, ManifestGroup):
+            # Groups contain a metadata document and the name of sub-groups/arrays
+            yield "zarr.json"
+            for member_name in node._members.keys():
+                yield member_name
+        elif isinstance(node, ManifestArray):
+            # TODO: Support listing when using other chunk_key_encodings
+            if (
+                separator := getattr(
+                    node.metadata.chunk_key_encoding, "separator", None
+                )
+                != "."
+            ):
+                raise NotImplementedError(
+                    f"Array listing only supports '.' as chunk key separator, "
+                    f"got {separator!r}"
+                )
```

Comment:
I think this is _slightly_ simpler:

```suggestion
        # TODO: Support listing when using other chunk_key_encodings
        elif (
            separator := getattr(node.metadata.chunk_key_encoding, "separator", None)
            != "."
        ):
            raise NotImplementedError(
                f"Array listing only supports '.' as chunk key separator, "
                f"got {separator!r}"
            )
        else:
 ```